### PR TITLE
config(engine): imitate adjust DM source config

### LINKED ISSUE #5535: dm on dataflow engine e2e test failed
### Which jobs are flaking?

DM-workflow

### Which test(s) are flaking?

TestDMJob

### Jenkins logs or GitHub Actions link

https://github.com/pingcap/tiflow/runs/6553393749?check_suite_focus=true

```go
[2022/05/23 10:33:56.486 +00:00] [INFO] [sync_worker.go:51] ["init sync worker"]
[2022/05/23 10:33:56.486 +00:00] [INFO] [broker.go:176] ["Using local storage with path"] [path=039a887e-2223-4e4a-98c6-7f623ab212ac/test/mysql-replica-01]
[2022/05/23 10:33:56.488 +00:00] [WARN] [task_runner.go:256] ["Task init returned error"] [id=93f7b638-d373-4c31-960d-ab1bb465ed11] [error="panic: empty flavor with empty gtid is invalid"] [errorVerbose="panic: empty flavor with empty gtid is invalid\ngithub.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).onNewTask.func2.1\n\tgithub.com/pingcap/tiflow/engine/executor/worker/task_runner.go:225\nruntime.gopanic\n\truntime/panic.go:838\ngithub.com/pingcap/tiflow/dm/pkg/gtid.MustZeroGTIDSet\n\tgithub.com/pingcap/tiflow/dm/pkg/gtid/gtid.go:70\ngithub.com/pingcap/tiflow/dm/pkg/binlog.MustZeroLocation\n\tgithub.com/pingcap/tiflow/dm/pkg/binlog/position.go:256\ngithub.com/pingcap/tiflow/dm/syncer.NewRemoteCheckPoint\n\tgithub.com/pingcap/tiflow/dm/syncer/checkpoint.go:385\ngithub.com/pingcap/tiflow/dm/syncer.NewSyncer\n\tgithub.com/pingcap/tiflow/dm/syncer/syncer.go:274\ngithub.com/pingcap/tiflow/engine/dm.(*syncWorker).InitImpl\n\tgithub.com/pingcap/tiflow/engine/dm/sync_worker.go:67\ngithub.com/pingcap/tiflow/engine/lib.(*DefaultBaseWorker).Init\n\tgithub.com/pingcap/tiflow/engine/lib/worker.go:212\ngithub.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).onNewTask.func2\n\tgithub.com/pingcap/tiflow/engine/executor/worker/task_runner.go:230\ngithub.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).onNewTask.func3\n\tgithub.com/pingcap/tiflow/engine/executor/worker/task_runner.go:255\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x2a755b9]

goroutine 1220 [running]:
github.com/pingcap/tiflow/engine/dm.(*unitHolder).close(0x0)
        github.com/pingcap/tiflow/engine/dm/unit.go:210 +0x19
github.com/pingcap/tiflow/engine/dm.(*syncWorker).CloseImpl(0xc000d7d068?, {0x3110e20?, 0xc001a6a920?})
        github.com/pingcap/tiflow/engine/dm/sync_worker.go:92 +0x1d
github.com/pingcap/tiflow/engine/lib.(*DefaultBaseWorker).Close(0xc0006e5a40, {0x39cad20?, 0xc001a5b2d8?})
        github.com/pingcap/tiflow/engine/lib/worker.go:349 +0x4e
github.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).onNewTask.func3.1()
        github.com/pingcap/tiflow/engine/executor/worker/task_runner.go:243 +0x58
github.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).onNewTask.func3()
        github.com/pingcap/tiflow/engine/executor/worker/task_runner.go:257 +0x36b
created by github.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).onNewTask
        github.com/pingcap/tiflow/engine/executor/worker/task_runner.go:238 +0x5db
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #5535

ref #5504 

### What is changed and how it works?

convert between and call existing functions.

#### Tests 

 - Integration test

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/jobmaster/dm/config/config.go [hunk 1]
```diff
@@ -166,16 +197,26 @@ func (c *JobCfg) fromDMTaskCfg(dmTaskCfg *dmconfig.TaskConfig) error {
 }
 
 func (c *JobCfg) adjust() error {
-	dmTaskCfg, err := c.toDMTaskCfg()
+	for _, upstream := range c.Upstreams {
```
Comment (GMHDBJD) [inline engine/jobmaster/dm/config/config.go:200]: I prefer move theses lines to L183

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by maxshuang — APPROVED** [review]

**Comment by lance6716** [comment]
conversion
/run-dm-integration-test

**Pull request merged**